packages/terra-data-grid/src/DataGrid.jsx

```diff
@@ -362,7 +276,7 @@ const DataGrid = injectIntl((props) => {
     );
 
     // Identify index of the active element in the DOM excluding data grid children
-    const index = focusableElements.indexOf(grid.current);
+    const index = focusableElements.indexOf(gridContainerRef.current);
```

Comment:
About this: The selector was trying to find `[tabindex]:not([tabindex='-1'])` as part of the query. However with this new implementation, the `tabindex={0}` is now on the DataGrid container (and not on the table). So we have to change the reference from the grid to the container.

Reply:
Then we should also update the focusableElements logic to use the proper container in the check